### ISSUE TITLE
Mixtures in Modelica

### ISSUE THREAD

**Issue by jowr** [CoolProp/ExternalMedia#5]
https://github.com/CoolProp/CoolProp/issues/263
Create a new interface on the Modelica level that allows for both two-phase conditions and mixtures. It might be possible to reuse some code from https://github.com/jowr/REFPROP2Modelica

**Comment by thorade** [comment]
Or wait for the New media interface presentation at the coming modelica
conference.
Am 16.07.2015 6:48 nachm. schrieb "Jorrit Wronski" <notifications@github.com

**Comment by thorade** [comment]
Or wait for the new Media interface presentation at the Modelica 2015 conference.

Related: 
https://trac.modelica.org/Modelica/ticket/1419
http://cordis.europa.eu/project/rcn/109104_en.html

**Comment by thorade** [comment]
The following link should work within the next couple of days:
http://dx.doi.org/10.3384/ecp15118653

Until then,  read pages 653-662 from this pdf:
https://www.modelica.org/events/modelica2015/ProceedingsOfThe11thInternationalModelicaConference.pdf

**Comment by jowr** [comment]
Thanks. It is a little sad that they mention the insufficient performance of external libraries using REFPROP as an example. The main drawbacks are summarised as 
1. Error handling when calling properties outside their validity region.
2. Limited support for partial derivatives.
3. Lack of support to speed up iterative calculation by providing good start values.
4. No access to the used tolerances, which may cause numerical problems when creating numerical derivatives. 
5. Non converging regions. 

... and CoolProp might be able to improve the performance and definitely supports many derivatives, see point 2.

**Comment by thorade** [comment]
This may be due to the fact that some of the project partners do also sell commercial fluid properties libraries. Also, my experience with Refprop developers is that they are very happy about feedback and quickly try to solve problems (as far as their codebase allows), so I hope the autors did provide good feedback in addition to writing a general complaint in the paper.

UPDATE: At the time the project started, CoolProp was pure-fluids only.

**Comment by ibell** [comment]
Yes, that has been my experience with REFPROP developers too :)  Perhaps
they emailed Eric Lemmon about their issues, but probably they didn't.

Ian

**Comment by thorade** [comment]
The Modelon presentation from the Modelica conference is available here:
http://www.slideshare.net/Modelon/multi-phase-mixture-media

**Comment by jowr** [comment]
In the slides they say that it is open-source. Do they refer to this repository? https://github.com/jwindahlModelon/MultiPhaseMixtureMedia

**Comment by thorade** [comment]
Yes, that is the library as presented at the conference. As I understand it, it was once planned that this could go straight into the Modelica Standard Library, but now it is more "another suggestion how to do it" and more discussion e.g. at the Modelica Design Meeting is necessary. Maybe the Modelica Language Specification has to be adapted in order to make a clean solution possible.

**Comment by thorade** [comment]
Also read this related blog post:
http://www.modelon.com/blog/articles/bringing-modelica-to-the-process-industry/
